infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/local/LocalCheInfrastructureProvisioner.java

```diff
@@ -82,56 +88,93 @@ public LocalCheInfrastructureProvisioner(InstallerConfigApplier installerConfigA
     }
 
     @Override
-    public void provision(Environment envConfig,
+    public void provision(EnvironmentImpl envConfig,
                           DockerEnvironment internalEnv,
                           RuntimeIdentity identity)
             throws InfrastructureException {
         String devMachineName = getDevMachineName(envConfig);
         if (devMachineName == null) {
             throw new InfrastructureException("ws-machine is not found on installers applying");
         }
+        excludeDirsFromSnapshot(internalEnv);
+        addProjects(internalEnv, identity.getWorkspaceId(), devMachineName);
```

Comment:
I don't think that usage of `addProjects` and `addInstallers` here are clear enough because one add volumes, another one adds volumes and env variables. But you can not rework it because they are private methods.